### PR TITLE
Some basic support for uint{16,32,64} codegen in CPU inductor

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116809
* __->__ #116810

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by chenyang78 — APPROVED** [review]
LGTM. Thanks!

**Review by eellison — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

File: torch/_inductor/codegen/cpp.py [hunk 1]
```diff
@@ -59,6 +59,9 @@
     torch.int32: "int",
     torch.int16: "short",
     torch.int8: "signed char",
+    torch.uint64: "unsigned long",
```
Comment (kadeng) [inline torch/_inductor/codegen/cpp.py:62]: As far as I know, while in practice (on 64-bit systems) these types have the right bit-size, this is not guaranteed by the standard. Maybe it's better to use C++11 fixed width integer types? ( https://en.cppreference.com/w/cpp/types/integer )

Comment (ezyang) [reply torch/_inductor/codegen/cpp.py:62]: Yeah, I don't know why the other usages here weren't using the fixed width types lol

**Pull request closed without merging**